### PR TITLE
Add getGamePassProductInfo()

### PR DESCRIPTION
**Pull request by Regalijan**
Does same thing as `getProductInfo()` but for gamepasses

### CODE REVIEW

File: typings/index.d.ts [hunk 1]
```diff
@@ -1457,6 +1459,11 @@ declare module "noblox.js" {
      */
     function getProductInfo(asset: number): Promise<ProductInfo>;
 
+    /**
+     * ✅ Gets `info` of `gamepass` and caches according to settings.
+     */
+    function getGamePassProductInfo(gamepass: number): Promise<GamePassProductInfo>;
```
Comment (suufi) [inline typings/index.d.ts:1465]: functions are in alphabetical order

File: test/asset.test.js [hunk 2]
```diff
@@ -95,6 +95,17 @@ describe('Asset Methods', () => {
...
+    return getGamePassProductInfo(2919875).then((res) => {
+      return expect(res).toMatchObject({
+        Name: expect.any(String),
+        Description: expect.any(String),
+        Creator: expect.any(Object),
+        PriceInRobux: expect.nullOrAny(Number)
+      })
+    })
+  })
+
```
Comment (suufi) [inline test/asset.test.js:108]: test case for invalid gamepass ID

File: lib/asset/getGamePassProductInfo.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+const http = require('../util/http.js').func
+const cache = require('../cache')
+
+// Args
+exports.required = ['gamepass']
+
+// Docs
+/**
+ * ✅ Get the info of an gamepass.
+ * @category Assets
```
Comment (suufi) [inline lib/asset/getGamePassProductInfo.js:11]: just `Asset` not `Assets`

File: lib/asset/getGamePassProductInfo.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+ * @returns {Promise<GamePassProductInfo>}
+ * @example const noblox = require("noblox.js")
+ * const gamePassInfo = await noblox.getGamePassProductInfo(2919875)
+**/
+
+// Define
+function getGamePassProductInfo (gamepass) {
+  return new Promise((resolve, reject) => {
+    const httpOpt = {
+      url: `//apis.roblox.com/game-passes/v1/game-passes/${gamepass}/product-info`,
```
Comment (suufi) [inline lib/asset/getGamePassProductInfo.js:22]: is there any other documented endpoint for this?

**Review by suufi — CHANGES_REQUESTED** [review]

File: lib/asset/getGamePassProductInfo.js (hunk 4 above)
Comment (Wolftallemo) [reply lib/asset/getGamePassProductInfo.js:22]: This is what Roblox staff recommend, I'm unaware of any other endpoints outside of the economy api (which is also undocumented)
https://devforum.roblox.com/t/action-needed-sunsetting-apirobloxcom/1796897#marketplace-5

**Comment by Wolftallemo** [comment]
image

**Review by Neztore — APPROVED** [review]

**Pull request merged**